lib/motions/move-to-definition-motion.coffee

```diff
@@ -0,0 +1,40 @@
+_ = require 'underscore-plus'
+{Input} = require '../view-models/view-model'
+{SearchCurrentWord} = require './search-motion'
+
+module.exports =
+class MoveToDefinition extends SearchCurrentWord
+  constructor: (@editor, @vimState) ->
+    super(@editor, @vimState)
+
+    # FIXME: This must depend on the current language
+    # NOTE: The only modification here is the addition of the $
+    defaultIsKeyword = "[@$a-zA-Z0-9_\-]+"
+    userIsKeyword = atom.config.get('vim-mode.iskeyword')
+    @keywordRegex = new RegExp(userIsKeyword or defaultIsKeyword)
+
+    searchString = @getCurrentWordMatch()
+    @input = new Input(searchString)
```

Comment:
Related to this and your note about replacing the scan with super:

I've taken out the majority of the code, bringing it down to just the call to `super` and added in an additional parameter to allow the search to halt on the first result. This works great for almost all items, so thanks for the great suggestions! (this will probably get much more complicated for scope-level `gd`, so I think it makes sense to keep it in a separate file until that's built)

However, I still am not sure about the right approach for supporting variables that start with `$`. All but those type of variable pass the test! I can get it to work if I keep the constructor and the custom `getSearchTerm` to include the $ and escape it properly, but I'm not sure if there is a good way to make this work without those two functions! Thoughts?